core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/AllObjectsTarget.java

```diff
@@ -0,0 +1,126 @@
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.sail.shacl.AST;
+
+import java.util.UUID;
+
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.repository.sail.SailRepositoryConnection;
+import org.eclipse.rdf4j.sail.SailConnection;
+import org.eclipse.rdf4j.sail.shacl.ConnectionsGroup;
+import org.eclipse.rdf4j.sail.shacl.RdfsSubClassOfReasoner;
+import org.eclipse.rdf4j.sail.shacl.ShaclSail;
+import org.eclipse.rdf4j.sail.shacl.Stats;
+import org.eclipse.rdf4j.sail.shacl.planNodes.ExternalFilterIsObject;
+import org.eclipse.rdf4j.sail.shacl.planNodes.PlanNode;
+import org.eclipse.rdf4j.sail.shacl.planNodes.PlanNodeProvider;
+import org.eclipse.rdf4j.sail.shacl.planNodes.Sort;
+import org.eclipse.rdf4j.sail.shacl.planNodes.TrimTuple;
+import org.eclipse.rdf4j.sail.shacl.planNodes.UnBufferedPlanNode;
+import org.eclipse.rdf4j.sail.shacl.planNodes.Unique;
+import org.eclipse.rdf4j.sail.shacl.planNodes.UnorderedSelect;
+
+/**
+ * sh:targetObjectsOf
+ *
+ * @author Håvard Mikkelsen Ottestad
+ */
+public class AllObjectsTarget extends NodeShape {
+
+	AllObjectsTarget(Resource id, ShaclSail shaclSail, SailRepositoryConnection connection, boolean deactivated) {
+		super(id, shaclSail, connection, deactivated);
+	}
+
+	@Override
+	public PlanNode getPlan(ConnectionsGroup connectionsGroup, boolean printPlans,
+			PlanNodeProvider overrideTargetNode, boolean negateThisPlan, boolean negateSubPlans) {
+		assert !negateSubPlans : "There are no subplans!";
+		assert !negateThisPlan;
+
+		PlanNode select = new Unique(
+				new Sort(
+						new TrimTuple(
+								new UnorderedSelect(
+										connectionsGroup.getBaseConnection(),
+										null,
+										null,
+										null,
+										UnorderedSelect.OutputPattern.ObjectPredicateSubject),
+								0,
+								1)));
+
+		return connectionsGroup.getCachedNodeFor(select);
+	}
+
+	@Override
+	public PlanNode getPlanAddedStatements(ConnectionsGroup connectionsGroup,
+			PlaneNodeWrapper planeNodeWrapper) {
+
+		PlanNode select = new Unique(
+				new Sort(
+						new TrimTuple(
+								new UnorderedSelect(
+										connectionsGroup.getAddedStatements(),
+										null,
+										null,
+										null,
+										UnorderedSelect.OutputPattern.ObjectPredicateSubject),
+								0,
+								1)));
+
+		return connectionsGroup.getCachedNodeFor(select);
+	}
+
+	@Override
+	public PlanNode getPlanRemovedStatements(ConnectionsGroup connectionsGroup,
+			PlaneNodeWrapper planeNodeWrapper) {
+		PlanNode select = new Unique(
+				new Sort(
+						new TrimTuple(
+								new UnorderedSelect(
+										connectionsGroup.getRemovedStatements(),
+										null,
+										null,
+										null,
+										UnorderedSelect.OutputPattern.ObjectPredicateSubject),
+								0,
+								1)));
+
+		return connectionsGroup.getCachedNodeFor(select);
+	}
```

Comment:
Removed doesn't guarantee to only return this that have been removed, but must return everything that could potentially have been removed. So this implementation will be fine.